typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -1,89 +1,676 @@
+
+    const currentConfig = await this.read();
+    // Check if we need to deploy a new ISM
+    if (
+      // if custom ISM -> config, do a new deploy
+      typeof currentConfig === 'string' ||
+      // if config -> config, AND types are different, do a new deploy
+      currentConfig.type !== targetConfig.type ||
+      // unless the new ISM is a Routing or Pausable ISM, do a new deploy
+      (targetConfig.type !== IsmType.ROUTING &&
+        targetConfig.type !== IsmType.FALLBACK_ROUTING &&
+        targetConfig.type !== IsmType.PAUSABLE)
+    ) {
+      const contract = await this.deploy({
+        config: targetConfig,
+      });
+
+      this.args.addresses.deployedIsm = contract.address;
+      return [];
+    }
+
+    const targetIsmType = targetConfig.type;
+    const logger = this.logger.child({
+      destination: this.chainName,
+      ismType: targetIsmType,
+    });
+    const provider = this.multiProvider.getProvider(this.chainName);
+
+    logger.debug(`Updating ${targetIsmType} on ${this.chainName}`);
+
+    // get owner
+    const owner = await Ownable__factory.connect(
+      this.args.addresses.deployedIsm,
+      provider,
+    ).owner();
+    const targetOwner = extractOwnerAddress(targetConfig.owner);
+
+    // Pausable ISMs are ownable, like Routing ISMs
+    // Check if the resolved owner is different from the current owner
+    // Return an ownership transfer transaction if required
+    if (targetIsmType === IsmType.PAUSABLE) {
+      if (!eqAddress(targetOwner, owner)) {
+        const tx = createAnnotatedEthersV5Transaction({
+          annotation: 'Transferring ownership of pausable ISM...',
+          chainId: this.domainId,
+          to: this.args.addresses.deployedIsm,
+          data: Ownable__factory.createInterface().encodeFunctionData(
+            'transferOwnership(address)',
+            [targetOwner],
+          ),
+        });
+        return [tx];
+      }
+      // no owner diff == no tx to return
+      return [];
+    }
+    // else if it's a routing ISM - update the existing one
+
+    // filter for known domains
+    const { availableDomains } = this.filterRoutingIsmDomains({
+      config: targetConfig,
+    });
+    targetConfig.domains = availableDomains;
+
+    // get current mailbox address
+    const client = MailboxClient__factory.connect(
+      this.args.addresses.deployedIsm,
+      provider,
+    );
+    const mailboxAddress = await client.mailbox();
+
+    // if mailbox delta, deploy new routing ISM before updating
+    if (!eqAddress(mailboxAddress, this.args.addresses.mailbox)) {
+      const newIsm = await this.deployRoutingIsm({
+        config: targetConfig,
+        logger,
+      });
+
+      this.args.addresses.deployedIsm = newIsm.address;
+    }
+
+    const delta = await routingModuleDelta(
+      this.chainName,
+      this.args.addresses.deployedIsm,
+      targetConfig,
+      this.multiProvider,
+      this.factories,
+      this.args.addresses.mailbox,
+    );
```

Comment:
i've tried to remove as much ICA logic as I can but this helper method does make a call to `resolveOrDeployAccountOwner`

how far do we go in re-ICAing stuff like this?

pretty much every ism utils function requires doing the `resolveOrDeployAccountOwner` thing

Reply:
we can punt the full migration to later but its unnecessary complexity everywhere